GDPR-Phase2-Java/src/main/java/com/gdpr/phase2/encoder/Base64Url.java

```diff
@@ -0,0 +1,50 @@
+package com.gdpr.phase2.encoder;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+
+
+import java.util.regex.Pattern;
+
+public class Base64Url {
+
+    private Base64Url() {
+    }
+    private static final Base64Url instance = new Base64Url();
+    public static Base64Url getInstance() {
+        return instance;
+    }
+
+    private static final int ONE_BYTE = 8;
+    private static final Pattern p = Pattern.compile("[A-Z0-9]",Pattern.CASE_INSENSITIVE);
+    /**
+     * decodes a base64url encoded bitfield string
+     *
+     * @static
+     * @param {string} str - base64url encoded bitfield string to be decoded
+     * @return {string} - bitfield string
+     */
+    public static String decode(String str) {
+        /**
+         * should contain only characters from the base64url set
+         */
+        if(!p.matcher(str).matches()) {
+            // throw error
+        }
+
+        /**
+         * @Functional Description -
+         *            1. First convert the string to byte array (byte array might have negative integers
```

Comment:
This method decodes into octets, we're actually not interested in whether that represents an integer or not.  I think (although I'm not an expert in Java or this library) you should be able to ignore integer value of the returned octets and just `toBinaryString` the value directly and pad the left to `ONE_BYTE`.  So the code below would then look like this:
```java
byte [] decodedBytes = Base64.decodeBase64(str);

int len = decodedBytes.length;
StringBuilder bitField = new StringBuilder();
for(int i = 0; i < len; i++) {
    String strBits = Integer.toBinaryString(decodedBytes[i]);
    bitField.append(StringUtils.leftPad(strBits, Base64Url.ONE_BYTE, "0"));
}
return bitField.toString();
```

Reply:
I will check that once.

Reply:
There is a different in output coming for both the codes
String strBits = Integer.toBinaryString(decodedBytes[i]); -->   strBits = 11111111111111111111111111101010

String strBits = Integer.toBinaryString(decodedBytes[i]&0xFF) --> strBits = 11101010

where decodeBytes[i] = -22

Reply:
Hmmm, ok...  I guess if it works, then no need to change it.